crates/tabby/src/services/completion.rs

```diff
@@ -335,21 +346,28 @@ impl CompletionService {
             let prompt = self
                 .prompt_builder
                 .build(&language, segments.clone(), &snippets);
-            (prompt, Some(segments), snippets)
+            if use_crlf {
+                (prompt.replace("\r\n", "\n"), Some(segments), snippets)
+            } else {
+                (prompt, Some(segments), snippets)
+            }
         } else {
             return Err(CompletionError::EmptyPrompt);
         };
 
-        let text = self.engine.generate(&prompt, options).await;
-        let segments = segments.cloned().map(|s| s.into());
+        let mut text = self.engine.generate(&prompt, options).await;
+        if use_crlf {
+            let re = Regex::new(r"([^\r])\n").unwrap(); // Match \n that is preceded by anything except \r
+            text = re.replace_all(&text, "$1\r\n").to_string() // Replace with captured character and \r\n
+        }
```

Comment:
extract a function `override_text(text, use_crlf)`.

Please also document a bit why this regex is needed for replacement (instead of just `\n` -> `\r\n`)

Reply:
`text` seems to be too much general, I changed it to `generated`